cranelift-codegen/src/regalloc/diversion.rs

```diff
@@ -163,9 +177,94 @@ impl RegDiversions {
         self.current.remove(&value).map(|d| d.to)
     }
 
+    /// Reset the current diversion for the entry of an `Ebb`, based on the recorded content
+    /// available on `func.entry_diversions`.
+    pub fn at_ebb(&mut self, entry_diversions: &EntryRegDiversions, ebb: Ebb) {
```

Comment:
I'm not such a fan of this name, because it's not clear what it's doing.

It seems to: 1) clear the current diversions. 2) put the new ebb's entry diversions into the current map.

Maybe it should be called `switch_to()`, `switch_ebb()`, something like that?

Reply:
Cursors have a function named `at_top(mut self, ebb: ir::Ebb) -> Self`.
@sstangl would it be ok, if I were to rename this function `at_top`, to match with cursor naming?

Reply:
Not having any answer, I will merge these changes open a follow-up to do this renaming later.

Reply:
I opened #958 to address this issue.